db/revtree.go

```diff
@@ -81,19 +87,37 @@ func (tree RevTree) MarshalJSON() ([]byte, error) {
 				rep.BodyKeyMap[strconv.FormatInt(int64(i), 10)] = info.BodyKey
 			}
 		}
-		rep.Channels[i] = info.Channels
+
+		// for leaf revisions we'll store channel information
+		// there's some duplication here for the current rev
+		// (active channels are stored outside the revtree)
+		// but special casing this doesn't seem worthwhile
+		if len(info.Channels) > 0 {
```

Comment:
When do we (need to) populate info.Channels for non-leaf nodes?  (i.e. why can't we assume that it's a leaf when this is populated)

Reply:
If we can't assume that - we have channels.Branched in syncData.Flags that we can use to quickly check whether the document has any non-tombstoned leaves (and so bypass this work in the most common case).  Any way to leverage that?